router/http_test.go

```diff
@@ -232,6 +232,57 @@ func (s *S) TestWildcardRouting(c *C) {
 	assertGet(c, "http://"+l.Addr, "dev.foo.bar", "3")
 }
 
+func (s *S) TestPathRouting(c *C) {
+	srv1 := httptest.NewServer(httpTestHandler("1"))
+	srv2 := httptest.NewServer(httpTestHandler("2"))
+	defer srv1.Close()
+	defer srv2.Close()
+
+	l := s.newHTTPListener(c)
+	defer l.Close()
+
+	defRoute := addRoute(c, l, router.HTTPRoute{
+		Domain:  "foo.bar",
+		Service: "1",
+	}.ToRoute())
+	pathRoute := addRoute(c, l, router.HTTPRoute{
+		Domain:  "foo.bar",
+		Service: "2",
+		Path:    "/2/",
+	}.ToRoute())
+
+	discoverdRegisterHTTPService(c, l, "1", srv1.Listener.Addr().String())
+	discoverdRegisterHTTPService(c, l, "2", srv2.Listener.Addr().String())
+
+	// Check that traffic received at the path is directed to correct backend
+	assertGet(c, "http://"+l.Addr, "foo.bar", "1")
+	assertGet(c, "http://"+l.Addr+"/2/", "foo.bar", "2")
```

Comment:
It will be directed even without the trailing slash, I couldn't work out which behaviour was preferable so settled on directing both as it's maybe a little less surprising for some.
I can add a test here to demonstrate.